packages/realm-server/tests/index.ts

```diff
@@ -12,3 +12,29 @@ import './queue-test';
 import './realm-server-test';
 import './virtual-network-test';
 import './billing-test';
+
+// There is some timer that is preventing the node process from ending promptly.
+// This forces the test to end with the correct response code. Note than a
+// message "Error: Process exited before tests finished running" will be
+// displayed because of this approach.
+import QUnit from 'qunit';
+(QUnit as any).on(
+  'runEnd',
+  ({
+    testCounts,
+  }: {
+    testCounts: {
+      passed: number;
+      failed: number;
+      total: number;
+      skipped: number;
+      todo: number;
+    };
+  }) => {
+    if (testCounts.failed > 0) {
+      process.exit(1);
+    } else {
+      process.exit(0);
+    }
+  },
+);
```

Comment:
this forces our tests to exit promptly despite some timer that seems to be preventing the node process from exiting